src/utils/db.ts

```diff
@@ -0,0 +1,54 @@
+import Dexie from 'dexie';
+import { Theme } from '../redux/appSlice';
+
+import { Todo } from '../redux/todosSlice';
+import {createAsyncThunk} from "@reduxjs/toolkit";
+
+interface AppConfigData {
+  id: number;
+  mode: Theme.LIGHT | Theme.DARK | undefined;
+}
+
+export class ToDoDatabase extends Dexie {
+  todos!: Dexie.Table<Todo, string>;
+  appConfig!: Dexie.Table<AppConfigData, number>;
+
+  constructor(databaseName: string) {
+    super(databaseName);
+    this.version(1).stores({
+      todos: '&++id,text,completed,timestamp',
+      appConfig: '++id,mode',
+    });
+  }
+}
+
+export const db = new ToDoDatabase('todoDatabase');
+
+export const saveTodosState = async (state: Todo[]) => {
+  try {
+    await db.todos.clear();
+    await db.todos.bulkPut(state);
+  } catch (error) {
+    console.error('Error saving state to database:', error);
+  }
+};
+
+export const fetchTodos = createAsyncThunk<Todo[], void>('todos/fetchTodos', async () => {
+  const todos = await db.todos.toArray();
+  return todos;
+});
+
+export const saveTheme = async (
+  mode: Theme.LIGHT | Theme.DARK | undefined
+): Promise<void> => {
+  await db.appConfig.put({ id: 1, mode });
```

Comment:
I still don't understand why do you need an ID here. You don't have multiple copies of your app

Reply:
The id acts as a stable reference to consistently update the same record. Without it, every change might generate a new record, cluttering the database. Having an id ensures we maintain a single, most recent configuration in the table.